Frameworks/Core/ERExtensions/Sources/er/extensions/foundation/ERXCompressionUtilities.java

```diff
@@ -33,21 +33,11 @@
 	 * @return gzipped NSData
 	 */
 	public static NSData gzipInputStreamAsNSData(InputStream input, int length) {
-		try {
-			ERXRefByteArrayOutputStream bos = new ERXRefByteArrayOutputStream(length);
+		try (ERXRefByteArrayOutputStream bos = new ERXRefByteArrayOutputStream(length)) {
 			if (input != null) {
-				GZIPOutputStream out = new GZIPOutputStream(bos);
-				try {
+				try (GZIPOutputStream out = new GZIPOutputStream(bos)) {
 					ERXFileUtilities.writeInputStreamToOutputStream(input, true, out, false);
 				}
-				finally {
-					try {
-						out.finish();
```

Comment:
Finish() is called from DeflaterOutputStream.close() so it is already called automatically.